Nix cleanup

lucasbergman opened the pull request:
I've learned a few things having upstreamed some stuff to nixpkgs lately :rofl:

lucasbergman committed:
maint: Set formatter Nix flake output so `nix fmt` works

I've idly thought about making the formatter a derivation that runs alejandra
over Nix code, then rustfmt over Rust code, etc--so that just running `nix
fmt` in the root cleans up everything--but I haven't made that work yet.

lucasbergman committed:
maint: Some Nix flake cleanups

Specifically:

  1. Switch from the nixosModule flake output (which is deprecated) to
     nixosModules.default
  2. Remove "with lib;" it's convenient, but I've gotten feedback when
     upstreaming module changes that that shouldn't be a thing in
     nixpkgs anymore
  3. Add a little type checking to the NixOS module options
  4. Switch the default config to empty (but leave the example)
  5. Use the convention "let cfg = config.services.heracles," a thing
     that's super common in NixOS

zaphar commented:
I've approved this but I'm afraid you collided with some changes of my own. Can you rebase and fix the conflicts?

lucasbergman committed:
maint: Run nix stuff through the formatter

There are several choices for formatting Nix code, but IMHO alejandra is the
right kind of opinionated: <https://github.com/kamadorueda/alejandra>.

lucasbergman committed:
maint: NixOS module writes config to the Nix store

Instead of writing the config attrset to a YAML file to a fixed path in /etc,
this makes the NixOS module write to a (content-addressed) config file in the
Nix store. We just use builtins.toJSON since YAML is a superset of JSON.

lucasbergman commented:
> I've approved this but I'm afraid you collided with some changes of my own. Can you rebase and fix the conflicts?

Easy peasy. Done! Also, I added commit 94b05b1 which I don't _think_ is controversial, but feel free to reject it as Too Weird™️.
